swent-group4/sudden-bump
# Block user

# Block User Feature Implementation

## 🔄 Changes Overview
This PR implements blocking functionality, allowing users to block other users and prevent them from appearing in friend lists, recommendations, and chat features.

### 🚀 Key Features
- Added block user functionality in the repository layer
- Implemented UI for blocking users from contact page
- Enhanced friend recommendations to exclude blocked users
- Added comprehensive test coverage

### 💻 Technical Implementation
#### Repository Layer
- Added `blockUser` method to `UserRepository` interface
- Implemented blocking in `UserRepositoryFirestore` using transactions
- Updated friend recommendations to filter out blocked users

#### ViewModel Layer
- Added `blockUser` method to handle blocking operations
- Updated state management for blocked users
- Added methods to manage blocked users list

#### UI Layer
- Added block user option in contact page dropdown menu
- Implemented confirmation dialog before blocking
- Updated friend recommendations UI to exclude blocked users

### 🧪 Testing
Added comprehensive tests:
- `UserRepositoryFirestoreTest`:
  - `blockUser_shouldUpdateAllLists`
  - `blockUser_whenTransactionFails_shouldCallOnFailure`
  - Updated `getRecommendedFriends_shouldReturnRecommendedFriends`
- `UserViewModelTest`:
  - `testBlockUser`

### 📦 Dependencies
- Updated Work Runtime KTX from 2.7.1 to 2.10.0
- Updated compileSdk from 34 to 35

### 🔍 Testing Instructions
1. Navigate to a user's contact page
2. Click the more options menu (three dots)
3. Select "Block User"
4. Confirm the blocking action
5. Verify the user is removed from friends list and recommendations
6. Verify that the blocked user can't find the current user anymore
